src/main/java/org/zalando/nakadi/repository/kafka/ProducerSendCommand.java

```diff
@@ -0,0 +1,123 @@
+package org.zalando.nakadi.repository.kafka;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.netflix.hystrix.HystrixCommand;
+import com.netflix.hystrix.HystrixCommandGroupKey;
+import org.apache.kafka.clients.producer.Callback;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.errors.NetworkException;
+import org.apache.kafka.common.errors.NotLeaderForPartitionException;
+import org.apache.kafka.common.errors.TimeoutException;
+import org.apache.kafka.common.errors.UnknownServerException;
+import org.apache.kafka.common.errors.UnknownTopicOrPartitionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.zalando.nakadi.domain.BatchItem;
+import org.zalando.nakadi.domain.EventPublishingStatus;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+public class ProducerSendCommand extends HystrixCommand<BatchItem> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ProducerSendCommand.class);
+    private static final String GROUP_PREFIX = "broker-";
+
+    private final KafkaFactory kafkaFactory;
+    private final String topicId;
+    private final BatchItem batchItem;
+    private final long timeout;
+
+    protected ProducerSendCommand(final KafkaFactory kafkaFactory,
+                                  final String topicId,
+                                  final BatchItem batchItem,
+                                  final long timeout) {
+        super(HystrixCommandGroupKey.Factory.asKey(GROUP_PREFIX + batchItem.getBrokerId()), (int) timeout);
+        this.kafkaFactory = kafkaFactory;
+        this.topicId = topicId;
+        this.batchItem = batchItem;
+        this.timeout = timeout;
+    }
+
+    @Override
+    protected BatchItem run() throws Exception {
+        try {
+            final ProducerRecord<String, String> kafkaRecord = new ProducerRecord<>(
+                    topicId,
+                    KafkaCursor.toKafkaPartition(batchItem.getPartition()),
+                    batchItem.getPartition(),
+                    batchItem.getEvent().toString());
+
+            final Producer<String, String> producer = kafkaFactory.takeProducer();
+            try {
+                final NakadiCallback callback = new NakadiCallback();
+                producer.send(kafkaRecord, callback);
+                final Exception exception = callback.result.get(timeout, TimeUnit.MILLISECONDS);
```

Comment:
That means that commands executed one by one, or histryx will create several threads to process commands?
In case of second approach - how many threads? There are batches of 600 events...